client/app/queue/AssignToView.jsx

```diff
@@ -72,12 +79,51 @@ class AssignToView extends React.Component<Props, ViewState> {
       title: `Task assigned to ${this.props.isTeamAssign ? 'team' : 'person'}`
     };
 
+    if (isReassignAction) {
+      return this.reassignTask();
+    }
+
     return this.props.requestSave('/tasks', payload, successMsg).
       then(() => {
         this.props.setTaskAttrs(appeal.externalId, { status: 'on_hold' });
       });
   }
 
+  reassignTask = () => {
+    const {
+      task,
+      isTeamAssign
+    } = this.props;
+
+    const payload = {
+      data: {
+        task: {
+          reassign: {
+            assigned_to_id: this.state.selectedValue,
+            assigned_to_type: isTeamAssign ? 'Organization' : 'User'
+          }
+        }
+      }
+    };
+
+    let assignee = 'person';
+
+    this.options().forEach((opt) => {
+      if (opt.value === this.state.selectedValue) {
+        assignee = opt.label;
+      }
+    });
+    const successMsg = { title: `Task reassigned to ${assignee}` };
+
+    return this.props.requestPatch(`/tasks/${task.taskId}`, payload, successMsg).
+      then((resp) => {
+        const response = JSON.parse(resp.text);
+        const preparedTasks = prepareTasksForStore(response.tasks.data);
+
+        this.props.setTaskAttrs(task.externalAppealId, preparedTasks[task.externalAppealId]);
```

Comment:
This is kind of tricky. We actually need to make a lot of potential changes to the store. Since we updated child tasks as well, and created a new task.

Reply:
I've merged the change with how we index these tasks. This should be 
```suggestion
        preparedTasks.map((preparedTask) => {
          this.props.setTaskAttrs(preparedTask.uniqueId, preparedTask);
        }
```